frontend/cypress/e2e/talentsearch/search-workflows.cy.js

```diff
@@ -8,89 +11,312 @@ describe("Talent Search Workflow Tests", () => {
       aliasMutation(req, "createPoolCandidateSearchRequest");
     });
 
-    cy.visit("/en/search");
+    // select some dimensions to use for testing
+    cy.getSkills().then((allSkills) => {
+      cy.wrap(allSkills[0]).as("testSkill"); // take the first skill for testing
+    });
+    cy.getGenericJobTitles().then((allGenericJobTitles) => {
+      const testGenericJobTitle1 = allGenericJobTitles[0]; // take the first ID for testing matching
+      cy.wrap(testGenericJobTitle1).as("testGenericJobTitle1");
+      cy.wrap(testGenericJobTitle1.classification).as("testClassification1");
+
+      const testGenericJobTitle2 = allGenericJobTitles[1]; // take the second ID for testing rejection
+      cy.wrap(testGenericJobTitle2.id).as("testGenericJobTitleId2");
+      cy.wrap(testGenericJobTitle2.classification).as("testClassification2");
+    });
   });
 
-  const searchReturnsGreaterThanZeroApplicants = () => {
-    cy.findByRole("heading", {
-      name: /Results: [1-9][0-9]* matching candidate/i,
+  // calculate how the classification will be displayed in the UI
+  const classificationName = (classification) =>
+    `${classification.group}-${classification.level
+      .toString()
+      .padStart(2, "0")}`;
+
+  it("searches for a candidate with all the filters and submits a request", () => {
+    // have a unique ID to track the test objects with
+    const uniqueTestId = Date.now().valueOf();
+    cy.log(`Test run ${uniqueTestId}`);
+
+    const searchFindsMySingleCandidate = () => {
+      cy.findByRole("article", {
+        name: `Cypress Test Pool EN 1 ${uniqueTestId} (IT-01 Business Line Advisory Services)`,
+      }).within(() => {
+        cy.contains("There is 1 matching candidate in this pool");
+
+        cy.findByRole("button", { name: /Request Candidates/i })
+          .should("exist")
+          .and("be.visible")
+          .and("not.be.disabled");
+      });
+    };
+
+    const searchRejectsMySingleCandidate = () => {
+      cy.findByRole("article", {
+        name: `Cypress Test Pool 1 EN ${uniqueTestId} (IT-01 Business Line Advisory Services)`,
+      }).should("not.exist");
+    };
+
+    // create a test user to attach test candidates to
+    cy.loginByRole("admin").then(() => {
+      cy.getMe()
+        .its("id")
+        .then((adminUserId) => {
```

Comment:
Ah shoot, I'm sorry, I didn't clue in that we were chaining login commands. There shouldn't be any reason for this, as cypress does things to run individual cypress commands sychronously

So if this is needed, then I think that means we're creating our cypress commands in a non-standard way.

```suggestion
    cy.loginByRole("admin")
    cy.getMe()
      .its("id")
      .then((adminUserId) => {
```

Reply:
And we can keep chaining, but my recollection from looking at the docs is that tests usually look more serial, using what cypress provides to operate synchronously between blocks when possible

```suggestion
    # Setup at start of test.
    cy.getMe().as("me")

    cy.loginByRole("admin")
    cy.get("@me").then((adminUser) => {
```

Reply:
Yeah, I think I started totally synchronous and then went too far asynchronous and need to dial it back.  It's a head trip for sure.  I stripped out the `.then(() => {` chains ([reduce unnecessary chaining](https://github.com/GCTC-NTGC/gc-digital-talent/pull/4998/commits/2d71a3e562dd5e1613212605718d0e29cbabf054)) and the tests are still passing so that seems to be fine.  I think that's as far as I want to go for now.

I saw [this page](https://glebbahmutov.com/blog/avoid-cypress-pyramid-of-doom/) about refactoring pyramids of doom. That might be a good refinement sometime.  I really wish Cypress had the ability to get an array of aliases at the same time like `Promise.all` instead of relying on the values to magically appear on the `this` context.

I'm wondering about seeding well-known uuids for the standard users (admin, applicant, etc) so we could put the IDs back in the user fixture file.  That would save a bunch of `me` queries in the tests.  Another possible improvement some day.